chart/templates/statsd/statsd-deployment.yaml

```diff
@@ -104,8 +104,17 @@ spec:
             - "--statsd.mapping-config=/etc/statsd-exporter/mappings.yml"
           {{- end }}
           resources: {{- toYaml .Values.statsd.resources | nindent 12 }}
-          {{- with .Values.statsd.env }}
-          env: {{- toYaml . | nindent 12 }}
+          env:
+          {{- if .Values.statsd.env }}
+          {{- toYaml .Values.statsd.env | nindent 12 }}
+          {{- end }}
+          {{- if .Values.config.metrics.metrics_allow_list }}
+            - name: AIRFLOW__METRICS__METRICS_ALLOW_LIST
```

Comment:
We shouldn't inject these env vars just to cause a restart. Annotations is the "right way" to cause that behavior.

That said, this is just how the statsd world behaves by default. e.g. [deleteIdleStats in StatsD itself](https://github.com/statsd/statsd/blob/7c07eec4e7cebbd376d8313b230cea96c6571423/exampleConfig.js#L61C3-L63). I believe the equivalent is [ttl in statsd-exporter](https://github.com/prometheus/statsd_exporter/blob/58769c7b4d128e7ceae1cf8d893260ed5b5afa4d/README.md?plain=1#L442). You can probably set it with `statsd.overrideMappings`.

Reply:
Thank you for your advice.

Then, I'm wondering is it bad idea adding a default TTL value here, aimed at ensuring they expire after a specified period of no updates.
I think it is distracting if metrics are not expired after changing metrics_allow_list and metrics_block_list so it is needed for others.

Reply:
Not sure about defaulting it, but if you want to expose it as an optional config, go for it!